### PR TITLE
Features/hdf 217 switch networks

### PR DESCRIPTION
**Pull request by ernestpascual**
Added switch network function copying sushiswap implementation.

### CODE REVIEW

**Comment by schystz** [comment]
@ernestpascual can you change base branch to develop? Surprised to see so many file changes :D

**Comment by ernestpascual** [comment]
> @ernestpascual can you change base branch to develop? Surprised to see so many file changes :D

woops sorry about that.

**Comment by tracyarciaga** [comment]
@ernestpascual do we have a preview link for this?

File: src/components/NetworkModal/index.tsx
Comment (schystz) [inline src/components/NetworkModal/index.tsx:1]: @ernestpascual I noticed that there's already some modal components in the code such as `components/Modal` and `components/ModalViews`, did you already take a look at them? I was hoping we can reuse the existing modal components to have a standard UI across the site if possible.

File: src/components/ModalHeader.tsx
Comment (schystz) [inline src/components/ModalHeader.tsx]: @ernestpascual perhaps we can move this inside `src/components/NetworkModal/`? But please also check my other comment first - https://github.com/HaloDAO/halodao-interface/pull/5#discussion_r633323346

File: src/components/NetworkModal/index.tsx
Comment (schystz) [inline src/components/NetworkModal/index.tsx]: @ernestpascual is it possible to move this inside `constants/networks.ts`?

File: src/components/NetworkModal/index.tsx
Comment (schystz) [inline src/components/NetworkModal/index.tsx:53]: @ernestpascual is it possible to move this array inside `constants/networks.ts`?

File: src/components/Typography.tsx
Comment (schystz) [inline src/components/Typography.tsx]: @ernestpascual fyi, there's already existing a typography helper called `TYPE` located in `src/theme/index.tsx`. I think it's better to reuse that than introducing this new component, for consistency.

**Review by schystz — COMMENTED** [review]
@ernestpascual it looks like the preview URL is not generated because of the ESLint errors. But before fixing those please also check out my other comments to see first if fixing the ESLint errors are still relevant. Happy to jump on a call as well to clarify any points that I raised.

**Comment by schystz** [comment]
@ernestpascual also, switching to Ethereum doesn't seem to work. I can switch between BSC - Polygon (vice versa) but when switching back to Ethereum I don't get the Metamask pop-up. Please see attached video:

https://user-images.githubusercontent.com/1032707/118458678-966bf180-b72d-11eb-97fd-a6d1e0ce1c60.mov

File: src/components/NetworkModal/index.tsx
Comment (ernestpascual) [reply src/components/NetworkModal/index.tsx:1]: I used components/Modal in here actually @schystz, see reference on line 5

File: src/components/ModalHeader.tsx
Comment (ernestpascual) [reply src/components/ModalHeader.tsx]: i used the structure here as reference. But I can move it also inside https://github.com/sushiswap/sushiswap-interface/tree/c9f8bad7fbc813690797399671eebc62e1fd1bb2/src/components

File: src/components/NetworkModal/index.tsx
Comment (ernestpascual) [reply src/components/NetworkModal/index.tsx:53]: okay i can. i just copied the implementation here for reference: https://github.com/sushiswap/sushiswap-interface/blob/c9f8bad7fbc813690797399671eebc62e1fd1bb2/src/components/NetworkModal/index.tsx#L10

File: src/components/ModalHeader.tsx
Comment (schystz) [reply src/components/ModalHeader.tsx]: Looks like their code becomes more messy now :D 

Saw the other comment and I think this is better put under `/src/components/Modal` as I would like to avoid dumping all components under `/src/components` whenever possible

File: src/components/Web3Network/index.tsx
Comment (schystz) [inline src/components/Web3Network/index.tsx]: @ernestpascual rename to `NetworkModal`?

File: src/components/NetworkModal/index.tsx
Comment (schystz) [reply src/components/NetworkModal/index.tsx:1]: Oh okay got it! Looks good in that case

File: src/components/NetworkModal/ModalHeader.tsx
Comment (schystz) [inline src/components/NetworkModal/ModalHeader.tsx]: @ernestpascual can we fix this eslint warning?

File: src/constants/index.ts
Comment (schystz) [inline src/constants/index.ts]: @ernestpascual would this be better if put under `constants/network.ts` since it's network related & probably won't be used anywhere else unlike the constants defined here in this file?

File: src/functions/styling.ts
Comment (schystz) [inline src/functions/styling.ts]: @ernestpascual I think this is no longer required

File: src/functions/index.ts
Comment (schystz) [inline src/functions/index.ts]: @ernestpascual I think this is no longer required (if we also removed `/src/functions/styling.ts`)

**Review by schystz — COMMENTED** [review]
Hi @ernestpascual , few minor feedbacks. I think this will be the last round for me and probably be approving after the requested changes are implemented/discussed.

**Comment by schystz** [comment]
1 more thing @ernestpascual , shall we remove Fantom and xDai? Since we don't plan to support it for v1
Screen Shot 2021-05-18 at 11 58 52 AM

File: src/components/ModalHeader.tsx
Comment (schystz) [reply src/components/ModalHeader.tsx]: I see that this file has already been moved. So resolving...

**Review by schystz — APPROVED** [review]
Approving this PR, but pls take care of that last thing we discussed on Slack (about hiding networks that we don't plan to initially support) before you merge @ernestpascual

**Review by tracyarciaga — APPROVED** [review]
Approve

**Pull request merged**
